mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb

```diff
@@ -3814,6 +3876,116 @@ resource "google_container_cluster" "with_node_config" {
 `, clusterName)
 }
 
+func testAccContainerCluster_withNodeConfigReservationAffinity(clusterName string) string {
+	return fmt.Sprintf(`
+resource "google_container_cluster" "with_node_config" {
+  name               = "%s"
+  location           = "us-central1-f"
+  initial_node_count = 1
+
+  node_config {
+    machine_type    = "e2-medium"
+    disk_size_gb    = 15
+    disk_type       = "pd-ssd"
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/monitoring",
+      "https://www.googleapis.com/auth/compute",
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/logging.write",
+    ]
+    service_account = "default"
+    metadata = {
+      foo                      = "bar"
+      disable-legacy-endpoints = "true"
+    }
+    labels = {
+      foo = "bar"
+    }
+    tags             = ["foo", "bar"]
+    preemptible      = true
+
+    // Updatable fields
+    image_type = "COS_CONTAINERD"
+
+    reservation_affinity {
+      consume_reservation_type = "ANY_RESERVATION"
+    }
+  }
+}
+`, clusterName)
+}
+
+func testAccContainerCluster_withNodeConfigReservationAffinitySpecific(clusterName string) string {
+	return fmt.Sprintf(`
+
+resource "google_project_service" "compute" {
+  service = "compute.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "container" {
+  service = "container.googleapis.com"
+  disable_on_destroy = false
+  depends_on = [google_project_service.compute]
+}
+
+
+resource "google_compute_reservation" "gce_reservation" {
+  name = "gce-reservation"
+  zone = "us-central1-f"
+
+  specific_reservation {
+    count = 1
+    instance_properties {
+      machine_type     = "n1-standard-1"
+    }
+  }
+
+  specific_reservation_required = true
+  depends_on = [google_project_service.compute]
+}
+
+resource "google_container_cluster" "with_node_config" {
+  name               = "%s"
+  location           = "us-central1-f"
+  initial_node_count = 1
+
+  node_config {
+    machine_type    = "n1-standard-1"
+    disk_size_gb    = 15
+    disk_type       = "pd-ssd"
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/monitoring",
+      "https://www.googleapis.com/auth/compute",
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/logging.write",
+    ]
+    service_account = "default"
+    metadata = {
+      foo                      = "bar"
+      disable-legacy-endpoints = "true"
+    }
+    labels = {
+      foo = "bar"
+    }
+    tags             = ["foo", "bar"]
+
+    // Updatable fields
+    image_type = "COS_CONTAINERD"
+
+    reservation_affinity {
+      consume_reservation_type = "SPECIFIC_RESERVATION"
+      key = "compute.googleapis.com/reservation-name"
```

Comment:
It looks like the docs say this should be `googleapis.com/reservation-name` - are they incorrect? (It seems like they must be since your test is passing but I'm curious if you might know any details.)

https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/NodeConfig#ReservationAffinity

Reply:
Yeah from what I can see, that is wrong in the doc. If I try that value, I get an error saying:
![image](https://user-images.githubusercontent.com/54791876/185734272-c8e046a2-f7d6-404d-b3d9-45fc7f0cf4cf.png)
So it is being set correctly by the provider, but just invalid. It also seems to be more consistent with the rest of GCP to use the full api url like compute.googleapis.com

Reply:
b/243557575